### PR TITLE
ASLR settings little refactor, available as util mean.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libr/util/sys.c [hunk 1]
```diff
@@ -460,6 +460,37 @@ R_API int r_sys_chdir(const char *s) {
 	return r_sandbox_chdir (s)==0;
 }
 
+R_API bool r_sys_aslr(int val) {
+	bool ret = true;
+#if __linux__
+	static const char *rva = "/proc/sys/kernel/randomize_va_space";
+	char buf[2] = {0};
+	snprintf(buf, sizeof (buf), "%d\n", val != 0 ? 2 : 0);
+	int fd = open (rva, O_WRONLY);
```
Comment (radare) [inline libr/util/sys.c]: use r_sandbox apis otherwise you will be bypassing the sandbox

File: libr/util/sys.c [hunk 2]
```diff
@@ -460,6 +460,37 @@ R_API int r_sys_chdir(const char *s) {
 	return r_sandbox_chdir (s)==0;
 }
 
+R_API bool r_sys_aslr(int val) {
+	bool ret = true;
+#if __linux__
+	static const char *rva = "/proc/sys/kernel/randomize_va_space";
+	char buf[2] = {0};
```
Comment (radare) [inline libr/util/sys.c]: the string you are trying to set in here is 3 byte long, not 2, this is overflowing

File: libr/util/sys.c [hunk 3]
```diff
@@ -460,6 +460,37 @@ R_API int r_sys_chdir(const char *s) {
 	return r_sandbox_chdir (s)==0;
 }
 
+R_API bool r_sys_aslr(int val) {
+	bool ret = true;
+#if __linux__
+	static const char *rva = "/proc/sys/kernel/randomize_va_space";
```
Comment (radare) [inline libr/util/sys.c]: this string doesnt needs to be saved as static

File: libr/util/sys.c [hunk 4]
```diff
@@ -460,6 +460,37 @@ R_API int r_sys_chdir(const char *s) {
...
+	char buf[2] = {0};
+	snprintf(buf, sizeof (buf), "%d\n", val != 0 ? 2 : 0);
```
Comment (radare) [inline libr/util/sys.c]: missing space before (

File: libr/util/sys.c [hunk 5]
```diff
@@ -460,6 +460,37 @@ R_API int r_sys_chdir(const char *s) {
...
+	int fd = open (rva, O_WRONLY);
+	if (fd != -1) {
+		if (write (fd, buf, sizeof (buf)) != 2) {
```
Comment (radare) [inline libr/util/sys.c]: shoulkd be 3, and compare with sizeof(buf) not against this hardcoded 2

Comment (devnexen) [reply libr/util/sys.c]: I merely move the original code ... I was not treating as a C string but just to write to the file.

File: libr/util/sys.c (hunk 1 above)
Comment (devnexen) [reply libr/util/sys.c]: Yes good point.

File: libr/util/sys.c (hunk 2 above)
Comment (devnexen) [reply libr/util/sys.c]: Originally we write two bytes to the file but 3 works too ...

**Pull request merged**